py/dml/codegen.py

```diff
@@ -1975,10 +2001,10 @@ def stmt_hashforeach(stmt, location, scope):
 def foreach_constant_list(site, itername, lst, statement, location, scope):
     assert isinstance(lst, AbstractList)
     spec = []
-    try:
-        loop_stack.append('unroll')
+    unroll = UnrolledLoopContext()
+    with unroll:
```

Comment:
You could rephrase as `with UnrolledLoopContext() as unroll:`, if you let `__enter__` return `self`. I don't know if that's better or worse.

Reply:
I'd prefer not to.

Reply:
fair enough